### ISSUE TITLE
differences with functools.cached_property?

### ISSUE THREAD

**Issue by odigity** [dssg/dickens#3]
Thank you for this module - I added `classproperty` and `cachedclassproperty` to my utilities.

I'm curious - is there any difference between your `cachedproperty` and the `cached_property` that was added to `functools` in Python 3.8?

https://docs.python.org/3/library/functools.html#functools.cached_property

**Comment by jesteria** [comment]
Howdy. My understanding is that Python devs considered something more expansive, but you can review the `functools.cached_property` source – it's pretty comparable:

* Python: [functools.cached_property](https://github.com/python/cpython/blob/main/Lib/functools.py#L963)
* Dickens: [descriptors.cachedproperty](https://github.com/dssg/dickens/blob/master/src/descriptors.py#L9)

I'd say the differences are slight and have mainly to do with refinements added to Python since Dickens was initially conceived (for Python v3.5):

* functools uses the descriptor interface method `__set_name__` (added with Python v3.6) rather than depend upon the name of the wrapped function (the most common case)
* functools defines `__class_getitem__` (added with Python v3.7) to support type hints
* functools attempts to protect against use of the descriptor on atypical classes (lacking `__dict__`)

So, to summarize: edge use-cases and features added since v3.5 – all very reasonable – but also nothing too likely to bite you.

I was also happy to see the descriptor added in v3.8; though, in projects using Dickens (for its other descriptors at least), I've continued to use the Dickens version of `cachedproperty`, without visible functional difference.

Separately, Dickens _should_ likely account for these changes in Python. I'll add issues for this; though, I don't see anything too urgent.

Cheers.
